Tools/TestWebKitAPI/Tests/WTF/SuspendableWorkQueue.cpp

```diff
@@ -37,19 +38,19 @@ TEST(WTF_SuspendableWorkQueue, Suspend)
     Lock lock;
     int taskCount = 100;
     int completedTaskCount = 0;
-    bool allTasksCompleted = false;
     bool suspended = false;
+    // Main thread only.
     bool suspendCompletionHandlerCalled = false;
+    bool taskAfterResumeCompleted = false;
+
     auto queue = SuspendableWorkQueue::create("com.apple.WebKit.Test.simple");
     for (int i = 0; i < taskCount; i ++) {
         queue->dispatch([&]() mutable {
             Locker locker { lock };
-            suspended = false;
```

Comment:
If I understand correctly from your commit log, what would happen is that all tasks could finish running before the call to suspend() (or at least before the request to suspend is processed).

As a result, you are now updating suspended and allTasksCompleted in new tasks that are scheduled after the queue is resumed.

However, wouldn't it mean that we would still fail the following check below?
`EXPECT_LE(completedTaskCount, taskCount);`

Maybe we should:
1. Schedule some tasks
2. suspend
3. Schedule some more tasks
4. Wait until suspension is complete.
5. Make sure not all tasks were completed
6. Sleep a bit and make sure no more tasks complete
7. Resume the work queue
8. Make sure all tasks complete.

Reply:
Why would the check fail? This is `EXPECT_LE` not `EXPECT_LT`.
But the test flow sounds reasonable; I will make the change.

Reply:
You're right, I misread the EXPECT check.